### ISSUE TITLE
vertical slider strange behaviour when [set <value>( message is triggered by slider output

### ISSUE THREAD

**Issue by jyg** [pure-data/pure-data#1831]
Hi,
In order to catch mouse vertical delta-y movements when captured by a vertical slider, I used the following patch to reset slider to its  middle position, after each new value. While on pd 0.52.2 this behaves as expected, on pd 0.53.0, the graphical rendering is correct but the slider outputs wrong values, as if set message were inoperative.
Set message behaves as expected if triggered manually (not by [t b f] object). Adding a [pipe 0] in the loop doesn't change anything. 
![image](https://user-images.githubusercontent.com/1431894/202655000-d44a8a01-7f4d-4534-b06d-f477ec6caa02.png)

**Comment by reduzent** [comment]
I think it is not a timing issue, rather sliders seem to ignore 'set' message during mouse down. Only after mouse release the 'set' message has an effect on the next output value. You can test with:

```
[metro 100]
|
[100 (
|
[vsl]
```

**Comment by umlaeute** [comment]
i think this (@reduzent's example) is due to improved "GUI-communication minimisation".
that is: if the slider already shows the requested value, it is not sent to the gui again.

a quick workaround for this is to set the value to some *other* value before setting it to the target.

```
[metro 100]
|
[set 101, set 100(
|
[vsl]
```

this doesn't solve the original issue though.

**Comment by jyg** [comment]
> i think this (@reduzent's example) is due to improved "GUI-communication minimisation". that is: if the slider already shows the requested value, it is not sent to the gui again.
> 
> a quick workaround for this is to set the value to some _other_ value before setting it to the target.
> 
> ```
> [metro 100]
> |
> [set 101, set 100(
> |
> [vsl]
> ```
> 
> this doesn't solve the original issue though.

Unfortunately, this doesn't solve the issue in my case.

**Comment by reduzent** [comment]
I don't see a difference either  with `[set 101, set 100(` instead of `[ set 100(`

**Comment by jyg** [comment]
If I compare source code for pd versions 0.42.2 and 43.0, 
In g_slider.c, line 825,     `x->x_pos = x->x_val;` has been suppressed in pd 0.43.0.
If I recompile with this line inserted, slider behaves as expected.

**Comment by Ant1r** [comment]
> Why should this line be suppressed ? Performance issue ?

no, only I didn't though about this case and I over-simplified... Sorry.
#1832 will fix that, in the way you found.

**Comment by jyg** [comment]
> no, only I didn't think about this case and I over-simplified... Sorry. #1832 will fix that, in the way you found.

Ok. That's great.
Thanks.

**Comment by umlaeute** [comment]
@jyg: sideo

**Comment by umlaeute** [comment]
this issue will be automatically closed once the relevant #1804 is merged into `master`
